tests/unit/acquisition/test_optimizer.py

```diff
@@ -141,11 +145,29 @@ def test_continuous_optimizer(
     expected_maximizer: list[list[float]],
     optimizer: AcquisitionOptimizer,
 ) -> None:
-
     maximizer = optimizer(search_space, _quadratic_sum(shift))
     npt.assert_allclose(maximizer, expected_maximizer, rtol=1e-3)
 
 
+@random_seed
+@pytest.mark.parametrize(
+    "optimizer",
+    [
+        generate_continuous_optimizer(),
+        generate_continuous_optimizer(num_restarts=3),
+        generate_continuous_optimizer(sigmoid=True),
+        generate_continuous_optimizer(sigmoid=True, num_restarts=3),
+        generate_continuous_optimizer(sigmoid=True, num_restarts=1, num_initial_samples=1),
+    ],
+)
+def test_optimize_continuous_raises_for_impossible_optimization(
+    optimizer: AcquisitionOptimizer,
+) -> None:
+    search_space = Box([-1], [2])
+    with pytest.raises(ValueError):
+        optimizer(search_space, _jerky_function())
```

Comment:
this doesn't assess that the optimiser does the right number of restarts, do we have any way of testing that?